fossunited/foss_profiles/doctype/foss_user_profile/test_foss_user_profile.py

```diff
@@ -36,3 +39,37 @@ def test_add_profile(self):
         profile_exists = frappe.db.exists(USER_PROFILE, {"user": frappe_user.name})
 
         self.assertTrue(profile_exists)
+
+    def test_private_profile_access(self):
+        fake = Faker()
+        test_name = fake.name()
+        test_user = frappe.get_doc(
+            {
+                "doctype": "User",
+                "email": str(uuid.uuid4()) + "@fossunited.org",
+                "first_name": test_name.split(" ")[0],
+                "name": test_name,
+                "full_name": test_name,
+            },
+        ).insert()
+
+        private_profile = frappe.get_doc(
+            {
+                "doctype": USER_PROFILE,
+                "user": test_user.name,
+                "is_private": 1,
+                "username": fake.user_name(),
+            }
+        ).insert()
```

Comment:
```suggestion
        private_profile = frappe.get_doc(
            doctype=USER_PROFILE, username=test_user.username
        )
        frappe.db.set_value(
            USER_PROFILE,
            private_profile.name,
            "is_private",
            1
        )
```

The previous code was creating a completely new User Profile doctype. Please note that once a User document is added, frappe/fossunited auto-creates a User Profile using information from the User document.

What we need to be doing is updating the existing User Profile doctype to mark it as private and the above code makes that change.

The problem then is, how to check that the exception gets raised. I'm not sure what is happening when we call `private_profile.get_profile()` here and how it is raising the exception.